src/ThemeJsonExtractor.php

```diff
@@ -0,0 +1,268 @@
+				foreach ( $base_tree as $node_name => $node_data ) {
+					$array_to_translate = self::array_get( $node_data, $data_path, null );
+					if ( is_null( $array_to_translate ) ) {
+						continue;
+					}
+
+					foreach ( $array_to_translate as $item_key => $item_to_translate ) {
+						if ( empty( $item_to_translate[ $key ] ) ) {
+							continue;
+						}
+
+						$translation = $translations->insert( $context, $array_to_translate[ $item_key ][ $key ] );
+						$translation->addReference( $file );
+					}
+				}
+			} else {
+				$array_to_translate = self::array_get( $theme_json, $path, null );
+				if ( is_null( $array_to_translate ) ) {
+					continue;
+				}
+
+				foreach ( $array_to_translate as $item_key => $item_to_translate ) {
+					if ( empty( $item_to_translate[ $key ] ) ) {
+						continue;
+					}
+
+					$translation = $translations->insert( $context, $array_to_translate[ $item_key ][ $key ] );
+					$translation->addReference( $file );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Given a file path, reads it as a JSON file
+	 * and returns an array with its contents.
+	 *
+	 * Returns an empty array in case of error.
+	 *
+	 * Ported from the core class `WP_Theme_JSON_Resolver`.
+	 *
+	 * @param string $file_path Path to file.
+	 * @param array $context A valid context resource created with stream_context_create(). Optional.
+	 *
+	 * @return array Contents of the file.
+	 */
+	private static function read_json_file( $file_path, $context = null ) {
+		$config = array();
+		if ( $file_path ) {
+			$decoded_file = json_decode(
+				file_get_contents( $file_path, false, $context ),
+				true
+			);
+
+			$json_decoding_error = json_last_error();
+			if ( JSON_ERROR_NONE !== $json_decoding_error ) {
+				WP_CLI::debug( "Error when decoding {$file_path}", 'make-pot' );
+
+				return $config;
+			}
+
+			if ( is_array( $decoded_file ) ) {
+				$config = $decoded_file;
+			}
+		}
+		return $config;
+	}
+
+	/**
+	 * Returns a data structure to help setting up translations for theme.json data.
+	 *
+	 * array(
+	 *     array(
+	 *         'path'    => array( 'settings', 'color', 'palette' ),
+	 *         'key'     => 'key-that-stores-the-string-to-translate',
+	 *         'context' => 'translation-context',
+	 *     ),
+	 *     array(
+	 *         'path'    => 'etc',
+	 *         'key'     => 'etc',
+	 *         'context' => 'etc',
+	 *     ),
+	 * )
+	 *
+	 * Ported from the core class `WP_Theme_JSON_Resolver`.
+	 *
+	 * @return array An array of theme.json fields that are translatable and the keys that are translatable.
+	 */
+	private static function get_fields_to_translate() {
+		$context = stream_context_create(
+			array(
+				'http' => array(
+					'method'  => 'GET',
+					'header'  => 'Content-type: application/json',
+					'timeout' => '3', // To make sure it resolves in a reasonable timeframe.
+				),
+			)
+		);
+		// Using the WordPress.org SVN repo resolved to a 403.
```

Comment:
Adding a User-Agent header should fix the 403.

Reply:
I switched the remote get to using `Utils\http_request`, which in turn uses the `Requests` library. SVN works with that.